workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.ts

```diff
@@ -161,9 +163,63 @@ export class DynamicAuthorizationService {
    * @returns - {@link DeleteGroupResponse}
    */
   public async deleteGroup(deleteGroupRequest: DeleteGroupRequest): Promise<DeleteGroupResponse> {
-    throw new Error('Not implemented');
+    const { authenticatedUser, groupId } = deleteGroupRequest;
+    const metadata: Metadata = {
+      actor: authenticatedUser,
+      action: this.deleteGroup.name,
+      source: {
+        serviceName: DynamicAuthorizationService.name
+      },
+      requestBody: deleteGroupRequest
+    };
+
+    let identityPermissions: IdentityPermission[];
+
+    try {
+      const { data } = await this.getIdentityPermissionsByIdentity({
+        identityId: groupId,
+        identityType: 'GROUP'
+      });
+
+      identityPermissions = data.identityPermissions;
+
+      await this._groupManagementPlugin.setGroupStatus({
+        groupId,
+        status: 'delete_pending'
+      });
+    } catch (error) {
+      metadata.statusCode = 400;
+      await this._auditService.write(metadata, error);
+
+      throw error;
+    }
+
+    if (identityPermissions.length > 0) {
+      try {
+        await this.deleteIdentityPermissions({
+          authenticatedUser,
+          identityPermissions
+        });
+      } catch (error) {
+        metadata.statusCode = 400;
+        await this._auditService.write(metadata, error);
+        throw new RetryError(error.message);
+      }
+    }
+
+    try {
```

Comment:
@bhudnell : We need to remove the `delete_pending` status once everything is completed gone right?

Reply:
I'm not sure I understand, is status stored separately from group? I thought it's group attribute so as group is deleted it gets removed as well. @htztomic ?

Reply:
Per our discussion this will be resolved in subsequent PR; for now I'm going to add comment with ToDo.